### PR TITLE
Updates to the Compose Release Notes

### PR DESCRIPTION
**Pull request by dockertopia**
Updating Compose release notes with latest information.

What I did:
- Expanded description for each line, tracked down related issues and linked them;
- Every change in the log with a referenced issue went into Bug Fixes section, all others into a "Changes" section. 
- Added releases later than 2.3.4 inclusive. 
- Added cross-references(xref) to compose release notes(RN) pages in repos for each release/tag
- Added a section for a mixed bag of previous releases with xref to RN in Compose repo.

@usha-mandya tagging Nicolas instead of Ulysses having listed PRs authoring in mind, assuming this is better for reviewing said lines. Let me know if this is a good approach.

### CODE REVIEW

File: compose/release-notes.md [hunk 1]
```diff
@@ -7,6 +7,90 @@ redirect_from:
   - /release-notes/docker-compose/
 ---
 
+# 2.5.0
+(2022-04-29)
+
+### Bug Fixes
+- Fixing panic with `compose down` command when -p flag specified. Closes [#9353](https://github.com/docker/compose/issues/9353){:target="_blank" rel="noopener" class="_"}.
```
Comment (usha-mandya) [inline compose/release-notes.md:14]: @dockertopia Could you change these to the past tense form? Also, could you update `Closes` to 'Fixes' when we reference external issues?

File: compose/release-notes.md [hunk 2]
```diff
@@ -7,6 +7,90 @@ redirect_from:
...
+- Removing a container with no candidate now produces a warning instead of an error. Closes [#9255](https://github.com/docker/compose/issues/9255){:target="_blank" rel="noopener" class="_"}.
+- Recovering behavior for 'compose up -d' of recreating containers of compose file images with refreshed content. Closes [#9259](https://github.com/docker/compose/issues/9259){:target="_blank" rel="noopener" class="_"}.
+- Removing the "Deprecated" mentions from -i and -t options to run and exec commands. These options are on by default and in use. Closes [#9229](https://github.com/docker/compose/pull/9229#discussion_r819730788){:target="_blank" rel="noopener" class="_"}. 
+
+### Changes
+
+- Removing the "Deprecated" mention from the --filter flag, to keep consistency with other commands. 
+- Docker compose --status, --filter and --format flags documentation updates.
+- Dependencies update: Bumping github.com/spf13/cobra from 1.3.0 to 1.4.0. Cobra library no longer requires Viper and all of its indirect dependencies [See cobra's release page](https://github.com/spf13/cobra/releases){:target="_blank" rel="noopener" class="_"}.
+- Dependencies update: Bumping Buildx from v0.7.1 to v0.8.0.
```
Comment (glours) [inline compose/release-notes.md:81]:
```suggestion
- Dependencies update: Bumping buildx from v0.7.1 to v0.8.0.
```

**Review by glours — APPROVED** [review]
Some minors modifications. Thanks @dockertopia that's a great summary of the latest releases 🙏

File: compose/release-notes.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (dockertopia) [reply compose/release-notes.md:14]: Thanks for the keen eye in all of these, I'm still checking when to use code formatting, there are some differences in contexts or teams. Will be paying more attention to this.

File: compose/release-notes.md [hunk 3]
```diff
@@ -7,6 +7,90 @@ redirect_from:
...
 
+# 2.5.0
+(2022-04-29)
+
+### Bug Fixes
+- Fixing panic with `compose down` command when -p flag specified. Closes [#9353](https://github.com/docker/compose/issues/9353){:target="_blank" rel="noopener" class="_"}.
+- Pass newly created project as input to start services (`docker compose up`). Closes [#9356](https://github.com/docker/compose/issues/9356){:target="_blank" rel="noopener" class="_"}.
+- Including services declared under links in docker-compose file as implicit dependencies. Closes [#9301](https://github.com/docker/compose/issues/9301){:target="_blank" rel="noopener" class="_"}.
+- Changes to `docker-compose pull` command to respect defined policy: 1)skip services configured as `pull_policy: never` and 2)ignore those with an existing image and `pull_policy: missing`. Closes [#3660](https://github.com/docker/compose/issues/3660){:target="_blank" rel="noopener" class="_"}.
+- Error building project from resources is no longer ignored in order to prevent `down`panic. Closes [#9383](https://github.com/docker/compose/issues/9383){:target="_blank" rel="noopener" class="_"}.
```
[the remark this answers is not in the document]
Comment (dockertopia) [reply compose/release-notes.md:18]: :)

File: compose/release-notes.md [hunk 4]
```diff
@@ -7,6 +7,90 @@ redirect_from:
...
+
+
+### Changes
+- Introducing support for `cache_from`, `cache_to`, `no_cache` and `pull` attributes in the build section. These attributes allow forcing a complete rebuild from sources and checking with registry for images used. These changes provide the basis for offering `--no-cache` and `--pull` options for compose build (or equivalent) command down the line.
+- Introducing support of an `--ssh `flag for the `build` command from CLI and Compose file. Closes [#7025](https://github.com/docker/compose/issues/7025){:target="_blank" rel="noopener" class="_"}.
+- Fixed typo in ssh flag description. Related to [#7025](https://github.com/docker/compose/issues/7025){:target="_blank" rel="noopener" class="_"}.
+- Pinning Kubernetes dependencies to the same version as in Buildx.
+- Dependencies update: Bumping Buildx to v0.8.1. to fix possible panic on handling build context scanning errors.
```
[the remark this answers is not in the document]
Comment (dockertopia) [reply compose/release-notes.md:58]: Did a find/replace.

File: compose/release-notes.md [hunk 5]
```diff
@@ -7,6 +7,90 @@ redirect_from:
...
+- Pass the interactive flag from the Compose CLI to the Docker one to run exec command.
```
[the remark this answers is not in the document]
Comment (dockertopia) [reply compose/release-notes.md:59]: Added

File: compose/release-notes.md [hunk 6]
```diff
@@ -7,6 +7,90 @@ redirect_from:
...
+
+- Compose commands now take value of `COMPOSE_PROJECT_NAME` environmental variable into consideration. Closes [#9316](https://github.com/docker/compose/issues/9316){:target="_blank" rel="noopener" class="_"}.
+- Fixed issue of `compose down` command that when executed in contexts without any services started/resources to deleted war returning an error. Error was due to command trying to delete inexistent network. Closes [#9333](https://github.com/docker/compose/issues/9333){:target="_blank" rel="noopener" class="_"}.
+
+
+### Changes
+- Introducing support for `cache_from`, `cache_to`, `no_cache` and `pull` attributes in the build section. These attributes allow forcing a complete rebuild from sources and checking with registry for images used. These changes provide the basis for offering `--no-cache` and `--pull` options for compose build (or equivalent) command down the line.
+- Introducing support of an `--ssh `flag for the `build` command from CLI and Compose file. Closes [#7025](https://github.com/docker/compose/issues/7025){:target="_blank" rel="noopener" class="_"}.
```
[the remark this answers is not in the document]
Comment (dockertopia) [reply compose/release-notes.md:55]: Added.

File: compose/release-notes.md [hunk 7]
```diff
@@ -7,6 +7,90 @@ redirect_from:
...
+- Added ssh config to the build options when building an image from a `docker compose up` command. Fixes [#9338](https://github.com/docker/compose/issues/9338){:target="_blank" rel="noopener" class="_"}.
+- Added inspection to container checking if a TTY is required. Running services with `tty:true` specified now show console output. Fixes [#9288](https://github.com/docker/compose/issues/9288){:target="_blank" rel="noopener" class="_"}.
+
+For the full change log or additional information, check the [Compose repository 2.4.1 release page](https://github.com/docker/compose/releases/tag/v2.4.1){:target="_blank" rel="noopener" class="_"}.  
+
+
+## 2.4.0 
+(2022-04-1)
+
+### Bug Fixes
```
Comment (usha-mandya) [inline compose/release-notes.md:46]: PR looks really good @dockertopia. 

nit: Could you add a blank line after the headings?

**Review by usha-mandya — APPROVED** [review]

**Pull request merged**